### PR TITLE
Vtmpy Mutator for Hexagon

### PR DESCRIPTION
**Pull request by aankit-ca**
Hexagon - Finding expressions reducable to vtmpy+ Rearranging expressions to equivalent forms

### CODE REVIEW

**Comment by aankit-ca** [comment]
@dsharletg PTAL

File: src/HexagonOptimize.cpp [hunk 1]
```diff
@@ -678,6 +678,25 @@ class OptimizePatterns : public IRMutator {
                 expr = mutate(op->a + neg_b);
                 return;
             } else {
+
+                // No widening subtracts for the expressions of the form:
```
Comment (dsharletg) [inline src/HexagonOptimize.cpp]: I think the best way to handle this might be to add "implementations" of the widening vsub to hvx_64.ll and hvx_128.ll, and add patterns for them. It's a bit more code, but IMHO it's less error prone.

Also - seems like this change should may be in a separate PR? It seems independent of the vtmpy stuff.

**Comment by aankit-ca** [comment]
Hi @dsharletg 
I've created a new PR: [https://github.com/halide/Halide/pull/2050](https://github.com/halide/Halide/pull/2050) with only Vtmpy Mutator and simd_op_check tests for the same.
Thanks

**Comment by dsharletg** [comment]
So, should this PR be closed in favor of #2050?
